cobra/model_building/models.py

```diff
@@ -390,34 +452,86 @@ def evaluate(self, X: pd.DataFrame, y: pd.Series,
             Dataset containing the target of each observation.
         split : str, optional
             Split name of the dataset (e.g. "train", "selection", or "validation").
-        metric: Callable (function), optional
-            Function that computes an evaluation metric to evaluate the model's
-            performances, instead of the default metric (RMSE).
-            The function should require y_true and y_pred arguments.
-            Metric functions from sklearn can be used, for example, see
+        metric : Callable (function), optional
+            Function that evaluates the model's performance, by calculating a
+            certain evaluation metric.
+            If the metric is not provided, the default metric AUC is used for
+            evaluating the model.
```

Comment:
RMSE: OK.
Duplication: I decided to leave the documentation of "metric" in the models - unfortunately, yes, it's then still in 2 spots (both LinearRegressionModel.evaluate and LogisticRegressionModel.evaluate), but some of the explanation is dedicated to the type of the model - e.g. see below example) + I think the documentation of "metric" belongs more to the model (how you evaluate the model), rather than forward feature selection. So I removed it in forward feature selection, but left it in the 2 models, because of the deviations of the text specifically for that model:
```
Examples:
            - sklearn.metrics.r2_score(y_true, y_pred, *, sample_weight=None, multioutput='uniform_average')
            - overall_estimated_commission_earned(y_true, y_pred,
                                                  avg_prob_buy_if_err_lower_than_20K=0.25,
                                                  avg_prob_buy_if_err_higher_than_20K=0.05,
                                                  pct_commission_on_buy=0.05)
```